### ISSUE TITLE
3 - Allow user to enter Loan Received from Bank on Schedule C (STEP 1)

### ISSUE THREAD

**Issue by GreggMoreland** [fecgov/fecfile-web-app#1003]
Business Reason 
 

As a Committee Administrator, I will be able to enter a **”Loan Received From Bank”** transaction on Schedule C of a Form 3X report so that I can accurately report on my committee's transactions.

NOTE: This transaction will also create a "Loan Received from Bank" transaction on Schedule A. This transaction also requires a ‘Child’ transaction to record the “Loan Agreement” and optionally might also include a second ‘Child’ transaction to record one or more ‘Loan Guarantor’.

 Acceptance Criteria 
 
**Given** I am a Committee Administrator editing an open Form 3X report
**Then** Add a Loans and Debts
**Then** I can enter a transaction with the type “Loan Received From Bank"
**Then** I will be required to enter the following information (unless otherwise noted):

---

 Lender 
#
-	**LENDER TYPE** (with LOOKUP)
-Organization only
- **ORGANIZATION NAME** (Required)

---

# Address 
#
-	**STREET ADDRESS** (Required)
-	**APARTMENT, SUITE, ETC.** (Optional)
-	**CITY** (Required)
-	**STATE/PROVINCE** (Required)
-	**ZIP/POSTAL CODE** (Required)

# Loan Information 
#
-	**ORIGINAL AMOUNT** (AMT-12, Required)
-	**PAYMENT TO DATE** (AMT-12, Required) (Should be $0 and read-only)
-	**BALANCE OUTSTANDING** (AMT-12, Required) (Should be same as Original Amount and read-only)
-	**MEMO ITEM** (Optional)

# Terms 
#
-	**DATE INCURRED** (Required)
-	**DUE DATE** (BLANK and read-only, implementation deferred until #1156)
-	**INTEREST RATE** (BLANK and read-only, implementation deferred until #1156)
-	**SECURED** (YN response, Required)

# Additional information 
#
-	**NOTE OR MEMO TEXT** (Optional)

---
**Then** I will be able to:
- Cancel
- Save **(When this is Saved, a 'Loan Received from Bank' transaction is created on Schedule A - See #1005)**
- Add C1- Loan Agreement (See #1011)
- Add C2 – Loan Guarantor (will be a dead link until implementation  of #1012)

**And** when this is saved, the transaction will be saved on the .fec file with line number SC/10

Note: This cannot be saved without #1005 AND  #1011

 QA Note
- Payment to Date and Balance Outstanding will be uneditable, and auto-calculation implementation deferred until #1103.
- Note that Loan Due Date and Loan Interest Rate fields will _not_ be implemented with this ticket, and will be implemented with #1156.
- Button to "Add Loan Guarantor" will be a dead link until implementation of #1012.

 Dev Note 

- Note that Loan Due Date and Loan Interest Rate fields will _not_ be implemented with this ticket, and will be implemented with #1156.
- Button to "Add Loan Guarantor" will be a dead link until implementation of #1012.
- Payment to Date and Balance Outstanding will be uneditable, and auto-calculation implementation deferred until #1103.

- [x] Update schema to match [Spec Updates spreadsheet tab here]( https://docs.google.com/spreadsheets/d/1H1zZmpkhUrKqe25OShtQrgbH1TrhIOSxvI1ZOM-Fw9Y/edit#gid=0)  **Please note this is a consolidated specification.** 
- [x] Disable the "Just checking ..." memo code date pop-up by adding this line to the transaction type: override doMemoCodeDateCheck = false;

 WIREFRAMES
![1003 - Loan Received from Bank.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/8b4cbe13-694b-4948-b18d-695061987207)

**In-line validation for "Is this loan secured?"**
![1003 - Loan Received from Bank - Terms.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/1232f159-f72b-4e59-a24f-77cf6b20cd87)

### QA Notes ### 
 null
### DEV Notes ### 
 null
### Design ### 
 null

**Comment by MitchellTCG** [comment]
@AureliaKhorsand @GreggMoreland 
I just want to verify that if this is for organization only, please remove:
LAST NAME (Required)
FIRST NAME (Required)
MIDDLE NAME (Optional)
PREFIX (Optional)
SUFFIX (Optional)

from the Acceptance Criteria.

**Comment by MitchellTCG** [comment]
@AureliaKhorsand 
Let review this tomorrow
![1003 - Loan Received from Bank.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/723cc4c4-6e2d-4b72-9c89-839ed6e0e1c2)

![1003 - Loan Received from Bank – 1.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/a0c7ef5d-e752-4b2b-8bc6-d4b6f04d743c)

**Comment by MitchellTCG** [comment]
@AureliaKhorsand Updated
![1003 - Loan Received from Bank.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/97d5efa0-55a9-4dc3-a38c-fb784f150836)
![1003 - Loan Received from Bank – 2.png](https://images.zenhubusercontent.com/619e83beec2d9fadb74ad6d1/72ab5207-9e48-4dce-9629-239896df8fab)

**Comment by MitchellTCG** [comment]
@AureliaKhorsand 
Please review. I made the updates from 66 and posted the wires here.

**Comment by AureliaKhorsand** [comment]
@MitchellTCG wires approved.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

**Comment by WiseQA** [comment]
QA review verified "Loan Received from Bank" is created, editable and savable (after Step 2 completed).

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/400b99a9-7f13-4d2c-98c8-7642deaaf4a1)

Note:  Loan Received from Bank saved under "Loans and Debts" once Step 2 #1011 is completed and saved.

![image.png](https://images.zenhubusercontent.com/61ba01e428a658b4eb0ca758/a4c1cf2d-9d9e-4a13-94dc-4d8a1ff9dc67)

(NOTE:  Retested contact window popups functionality is working correctly)

**Comment by WiseQA** [comment]
QA review retested "Loan Received from Bank" and verified that it is created, editable and savable (after Step 2 completed).

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbGdTIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--a477dfe5016dc4f8658bec4aa14e32e023701521/image.png)

Verified "Loan" label has been added

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbU1TIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--763d5294ca16c9e4b25443239a56b47e9e3e2baa/image.png)

Verified text in text boxes were corrected per wireframe:

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbkVTIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--195adebbf59643034ef3fbf635ab5202fc5a2314/image.png)

QA Review Completed.   Moved to Stage Ready.
